app/controllers/languages_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# LanguagesController: for language functionality
+class LanguagesController < ApplicationController
+  before_action :authenticate_user!
+
+  # Returns all languagaes as an array of arrays where each inner is in the format
+  # ['eng', 'English', supported: {sms: true, phone: true, email: true}]
+  # Example Return: [["eng", "English", {}]...,["zho","Chinese", {}],["zul","Zulu", {}]]
+  def language_data
+    render json: Languages.all_languages
+  end
+
+  # Matches code to display name
+  # params[:language_codes] must be a valid array of language codes (no nils or invalids)
+  # Returns array of the display names for those codes
+  # PARAM EXAMPLE: ['oci', 'tgk']
+  # RETURN EXAMPLE: ['Occitan', 'Tajik']
+  def translate_language_codes
+    render json: { display_names: params[:language_codes].map { |lang_code| Languages.all_languages[lang_code.to_sym][:display] } }
```

Comment:
Seeing an issue coming from here. Steps to reproduce:
1. On branch `1.28.0`
2. Enroll a monitoree with primary language set to `'Engish'` (typo intended)
3. Switch to this branch and run the migration
4. Visit the patient page for the monitoree

This results in a toast 500 error, with the following server output:

<img width="770" alt="image" src="https://user-images.githubusercontent.com/8235974/115927045-d652f800-a451-11eb-85bd-854462541e23.png">

And based on the following (post migration), it seems like something may have gone wrong with the migration. Shouldn't the `primary_language` be `'eng'"?

<img width="659" alt="image" src="https://user-images.githubusercontent.com/8235974/115927219-15814900-a452-11eb-8c98-a3c7aa8a5405.png">

Reply:
Great catch here! I've updated the way this logic works, because this was a bug. Thanks Tom!